### PR TITLE
Add obfuscation toggle for private key input

### LINKED ISSUE #20630: Allow user to toggle obfuscation on input of private key
### Describe the bug

When attempting to import a private key to extension, each character is obfuscated for security.  If the user makes a typing error it is impossible to review and correct the input.  This results in any miskey requiring the user to clear the field and start over from the beginning.  It would be nice to be able to enable and disable obfuscation on this field to help support any user that is manually entering a key.

### Steps to reproduce

1. Tap the accounts menu drop-down
2. Select `Import account`
3. manually input a private key with the keyboard

### Error messages or log output

_No response_

### Version

10.34.5

### Build type

None

### Browser

Chrome

### Operating system

Windows

### Hardware wallet

_No response_

### Additional context

https://recordit.co/4Zy4kXAjpi

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/MetaMask/metamask-extension/issues/20630

## Explanation
Adds a show/hide element which toggles whether the private key input value is visible or not.

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/d0b2da48-fed9-403f-aaf3-907b4336b136

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/4d8608fc-f765-4b45-a89b-edb13bfd9542

## Manual Testing Steps
1. Click the accounts menu dropdown
2. Select `Import account`
3. Ensure the private key input defaults to hidden
4. Ensure the show/hide toggle switches between input visibility appropriately
5. Ensure that the private key import functionality is otherwise unchanged

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by HowardBraham** [comment]
This looks visually unbalanced.  I think this would look better with the eye button inside the text field, like this:

image

**Comment by plasmacorral** [comment]
Would still be nice to have this even though this was closed by the stalebot.

**Comment by dawnseeker8** [comment]
i have updated the branch with latest code and also update the code to match latest change. now the PR is ready for review https://github.com/MetaMask/metamask-extension/pull/20719

**Comment by dawnseeker8** [comment]
> This looks visually unbalanced. I think this would look better with the eye button inside the text field, like this:
> 
> image

Let me do something to achieve this to move the eye button inside the box.

**Comment by dawnseeker8** [comment]
> > This looks visually unbalanced. I think this would look better with the eye button inside the text field, like this:
> > image
> 
> Let me do something to achieve this to move the eye button inside the box.

To move the eye inside the box is not easy since how we structure the FormTextField, i have created a new `form-password-field` which have use most of `form-text-field` styling. the final field look like this .

Screenshot 2024-02-22 at 18 16 28

**Comment by dawnseeker8** [comment]
@ryanml Can you review this ticket when you have time? thank you very much!

**Comment by dawnseeker8** [comment]
@HowardBraham  hi, Can you do a review of this PR when you have time? thank you very much, we will like to close this issue ASAP.

**Comment by HowardBraham** [comment]
@georgewrmarshall should look at this with respect to the new Design System stuff.

Also... this PR is currently blocked by the unsolved mystery of why all PRs from external contributors fail on `prep-deps`.

**Comment by ryanml** [comment]
@dawnseeker8 thanks for the updates!

File: app/_locales/en/messages.json
Comment (georgewrmarshall) [inline app/_locales/en/messages.json:3605]: Hey @coreyjanssen, could you please check the content here some of it is for the aria label for screen readers

File: ui/components/multichain/import-account/private-key.js
Comment (georgewrmarshall) [inline ui/components/multichain/import-account/private-key.js:35]: You can create a `FormPasswordField` if you like but you could also achieve this by including the `ShowHideToggle` button using the `FormTextField` => `textFieldProps` => `endAccessory` prop

```jsx
import {
  FormTextField,
  TextFieldSize,
  TextFieldType,
} from '../../component-library';

const [showPrivateKey, setShowPrivateKey] = useState(false);

<FormTextField
  id="private-key-box"
  size={TextFieldSize.Lg}
  autoFocus
  type={showPrivateKey ? TextFieldType.Text : TextFieldType.Password}
  textFieldProps={{
    endAccessory: (
      <ShowHideToggle
        shown={showPrivateKey}
        id="show-hide-private-key"
        title={t('privateKeyShow')}
        ariaLabelShown={t('privateKeyShown')}
        ariaLabelHidden={t('privateKeyHidden')}
        onChange={() => setShowPrivateKey(!showPrivateKey)}
      />
    ),
  }}
```


https://github.com/MetaMask/metamask-extension/assets/8112138/060db4aa-9e7e-4eb8-a981-ba9d6c57e953

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! I think if we are to create a `FormPasswordField` it should live in `ui/components/app` folder. I've provided an example of using the `endAccessory` prop to achieve the button within the input also.

Another reason to create a reusable component is to improve the consistency for all password type inputs. Here is one that should use the same technique. Could we create some tickets to replace other instances if this new password component gets merged? 


https://github.com/MetaMask/metamask-extension/assets/8112138/71b04498-f784-43a6-a6c0-cdd934da115b

File: ui/components/multichain/import-account/private-key.js
Comment (dawnseeker8) [reply ui/components/multichain/import-account/private-key.js:35]: Thank you very much @georgewrmarshall , i will use this props to redo the `private-key.js` and remove the extra `form-password-field` to reduce the complexity. :)

File: app/_locales/en/messages.json
Comment (coreyjanssen) [reply app/_locales/en/messages.json:3623]: Looks good to me!

**Review by coreyjanssen — DISMISSED** [review]

File: ui/components/multichain/import-account/private-key.js
Comment (dawnseeker8) [reply ui/components/multichain/import-account/private-key.js:35]: hi, @georgewrmarshall i have updated the `private-key.js` to use `form-text-field` with `endAccessory` attribute, please when you have time, review again the file and `resolve` this conversation.  thank you very much.

File: ui/components/ui/show-hide-toggle/index.scss
Comment (georgewrmarshall) [inline ui/components/ui/show-hide-toggle/index.scss:9]: Adding display flex here updates the component icon aligment

### Before

Screenshot 2024-03-05 at 7 25 57 PM

### After

Screenshot 2024-03-05 at 7 27 18 PM

**Review by georgewrmarshall — APPROVED** [review]
Hey @dawnseeker8, fantastic job! I've made a minor tweak to align the icon in the show-hide-toggle component and updated the before/after screencasts to match the code changes. Excellent work! 💯 LGTM!

File: ui/components/ui/show-hide-toggle/index.scss
Comment (dawnseeker8) [reply ui/components/ui/show-hide-toggle/index.scss:9]: thank you for the change. i will resolve this conversation.

**Review by garrettbear — APPROVED** [review]

**Comment by dawnseeker8** [comment]
have successfully ganed two review approver, however, need to ask somebody to fix CI pipeline or merge with without passing all action status.

**Comment by dawnseeker8** [comment]
Another same PR https://github.com/MetaMask/metamask-extension/pull/23371 has been created internally as workaround to run the CI. so close this ticket.

**Comment by dawnseeker8** [comment]
Another same PR https://github.com/MetaMask/metamask-extension/pull/23371 has been created internally as workaround to run the CI. so close this ticket.

**Pull request closed without merging**
